internal/token/token_csp_test.go

```diff
@@ -0,0 +1,51 @@
+package token
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestMultipleCSPRequests(t *testing.T) {
```

Comment:
I think we want another test for the behavior when:

* CSP is down (don't spam CSP)
* Scope is invalid (also don't spam CSP)

Maybe both of these should switch to a slower polling interval?
If they stop refreshing, they can't send metrics, so the buffer will build up... what happens?